### PR TITLE
Replace deprecated NewFakeClient usages with NewFakeClientWithScheme

### PR DESCRIPTION
**Pull request by racevedoo**
Hi guys, `NewFakeClient` is now deprecated. This PR replaces it with `NewFakeClientWithScheme`. If there is any better approach, let me know.

### CODE REVIEW

**Comment by barkbay** [comment]
Should we not take the opportunity to address #1274 instead ?

**Comment by sebgl** [comment]
@racevedoo one option could be to build our own wrapper around the fake client construction (`k8s.NewFakeClient(...)`). This function would setup the default scheme and return the initialized fake client. WDYT?

**Comment by racevedoo** [comment]
@sebgl I guess that's a better option. I'll provide some update soon

**Comment by racevedoo** [comment]
@sebgl just updated. I just have a couple of questions:
1) I added a "github.com/elastic/cloud-on-k8s/pkg/utils/test/fakek8sclient" package. Is this a good name? Is there a pattern used for tests through the project?
2) I got stuck in an import cycle, I'm not sure how to solve it. Ideas are welcome.
```
package github.com/elastic/cloud-on-k8s/pkg/controller/common/reconciler (test)
	imports github.com/elastic/cloud-on-k8s/pkg/utils/test/fakek8sclient
	imports github.com/elastic/cloud-on-k8s/pkg/controller/common/scheme
	imports github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1beta1
	imports github.com/elastic/cloud-on-k8s/pkg/controller/common
	imports github.com/elastic/cloud-on-k8s/pkg/controller/common/reconciler
```

**Comment by sebgl** [comment]
@racevedoo I checked out your PR locally to look at the import cycle. Some notes:

* About the **import cycle**
I think the problem is that we import `"github.com/elastic/cloud-on-k8s/pkg/controller/common"` in `apis/elasticsearch/v1beta1/elasticsearch_types.go`. I think we use it only for `common.Int32(1)`. We can either:
    * replace it by a `*int32` that does not rely on `common.Int32()` (we'll probably have to make a global int32 variable and reference its address).
    * move `common.Int32` to another package. IMHO it would make sense to have it in something like `utils/pointer/numeric.go`. Up to you if you want to do that in this PR, or move on with the first approach.

* About **naming**
`k8s.WrapClient(fakek8sclient.CreateFakeClientOrFatal(tt.args.initialObjects...))` feels a bit verbose to me. Some suggestions:
    * move `CreateFakeClientOrFatal` in `utils/k8s/fake.go`, so we can call it with the `k8s.` pkg prefix.
    * rename it to `FakeClient()`, so we can call `k8s.FakeClient()`
    * embed the call to `WrapClient()` so the function directly returns our own `k8s.Client` interface, and we don't need to call `k8s.WrapClient()`

Please let me know if you think that doesn't make sense. And thanks for your contribution!

**Comment by racevedoo** [comment]
@sebgl about the import cycle, I guess we can just replace it by a `*int32`, at least for now(maybe the `utils/pointer/numeric.go` can come on a later PR, this one is getting quite large) .

On naming, agreed. I'll update this PR soon

**Comment by racevedoo** [comment]
@sebgl done! I just had one doubt:

The `k8s.io/utils` package already provides functions to convert an int to a pointer, should we use it?

**Comment by sebgl** [comment]
> The k8s.io/utils package already provides functions to convert an int to a pointer, should we use it?

It looks like we already import it so yeah, let's go for it!

File: pkg/controller/common/deployment/reconcile_test.go [hunk 1]
```diff
@@ -7,17 +7,16 @@ package deployment
 import (
 	"testing"
 
+	"github.com/elastic/cloud-on-k8s/pkg/utils/pointer"
+
```
Comment (sebgl) [inline pkg/controller/common/deployment/reconcile_test.go]: nit: can you remove that empty line?

File: pkg/controller/license/license_controller_test.go [hunk 2]
```diff
@@ -173,7 +172,7 @@ func TestReconcileLicenses_reconcileInternal(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			require.NoError(t, v1beta1.AddToScheme(scheme.Scheme))
```
Comment (sebgl) [inline pkg/controller/license/license_controller_test.go]: I think we don't need that anymore

File: pkg/controller/kibana/driver_test.go [hunk 3]
```diff
@@ -198,7 +197,7 @@ func TestDriverDeploymentParams(t *testing.T) {
 			kb := tt.args.kb()
```
Comment (sebgl) [inline pkg/controller/kibana/driver_test.go]: there's a `kbtype.SchemeBuilder.AddToScheme(s); ` above that we dont need anymore

File: pkg/controller/common/deployment/reconcile_test.go (hunk 1 above)
Comment (racevedoo) [reply pkg/controller/common/deployment/reconcile_test.go]: will do!

Comment (racevedoo) [reply pkg/controller/common/deployment/reconcile_test.go]: done!

File: pkg/controller/kibana/driver_test.go (hunk 3 above)
Comment (racevedoo) [reply pkg/controller/kibana/driver_test.go]: done!

File: pkg/controller/license/license_controller_test.go (hunk 2 above)
Comment (racevedoo) [reply pkg/controller/license/license_controller_test.go]: yep, removed

**Comment by racevedoo** [comment]
@sebgl just updated. The change to use `k8s.io/utils/pointer` will come on another PR.

**Comment by sebgl** [comment]
Jenkins test this please

**Comment by sebgl** [comment]
CI fails with:
```
 Error: file(s) without license header:
08:59:14  ./build/../pkg/utils/k8s/fake.go
08:59:14  ./build/../pkg/utils/pointer/numeric.go
```

@racevedoo can you add the license header on top of both files (the same one that exists on other files).

**Comment by racevedoo** [comment]
@sebgl done! completely forgot :stuck_out_tongue_closed_eyes:

**Comment by sebgl** [comment]
Jenkins test this please

**Pull request merged**